test/realm_object_test.dart

```diff
@@ -68,6 +69,20 @@ class _BoolValue {
   late bool value;
 }
 
+extension on DateTime {
+  String toNormalizedDateString() {
+    final utc = toUtc();
```

Comment:
Why this formatting function changes the date? If the date is local and you want to format it will return utc date.

Reply:
This is used for tests only - to verify that the string output by Core matches our expectations. Since Core dates are always utc, it makes sense to always convert it.